lib/griddler/mailgun/adapter.rb

```diff
@@ -50,8 +50,8 @@ def headers
 
       def extract_headers
         extracted_headers = {}
-        if params['message-headers']
-          parsed_headers = JSON.parse(params['message-headers'])
+        if params['message_headers']
+          parsed_headers = JSON.parse(params['message_headers'])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.